lib/manager/jsonnet-bundler/artifacts.ts

```diff
@@ -0,0 +1,102 @@
+import { quote } from 'shlex';
+import { TEMPORARY_ERROR } from '../../constants/error-messages';
+import { logger } from '../../logger';
+import { ExecOptions, exec } from '../../util/exec';
+import { readLocalFile } from '../../util/fs';
+import { getRepoStatus } from '../../util/git';
+import { regEx } from '../../util/regex';
+import type {
+  PackageDependency,
+  UpdateArtifact,
+  UpdateArtifactsResult,
+} from '../types';
+
+function dependencyUrl(dep: PackageDependency): string {
+  const url = dep.lookupName;
+  if (dep.managerData?.subdir) {
+    return url.concat('/', dep.managerData.subdir);
+  }
+  return url;
+}
+
+export async function updateArtifacts(
+  updateArtifact: UpdateArtifact
+): Promise<UpdateArtifactsResult[] | null> {
+  const { packageFileName, updatedDeps, newPackageFileContent, config } =
+    updateArtifact;
+  logger.trace(`jsonnetfile.updateArtifacts(${packageFileName})`);
+  logger.trace({ newPackageFileContent });
+
+  const lockFileName = packageFileName.replace(regEx(/\.json$/), '.lock.json');
+  const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+
+  if (!existingLockFileContent) {
+    logger.debug('No jsonnetfile.lock.json found');
+    return null;
+  }
+
+  try {
+    const execOptions: ExecOptions = {
```

Comment:
I think we need to extract the `version` so we can use the right `jb` version.

https://github.com/jsonnet-bundler/jsonnet-bundler/releases/tag/v0.3.1

Reply:
I don't think so. The release notes state that it's v0.3.1 can handle the jsonnetfile.json files produced by 0.2.0.
```
The format of jsonnetfile.json has changed. While v0.3.0 can
handle the old v0.2 format, v0.2 can't and must not be used with a
jsonnetfile.json created by v0.3.0
```

I'm targeting jsonnet-bundler 0.4.0 anyway, since starting from that version you can update single dependencies. In pre-4.0.0, `jb update` would update all dependencies uncoditionally.

Reply:
Yes, please note this in the readme.